zebra/zebra_vty.c

```diff
@@ -976,47 +988,28 @@ DEFPY (show_ip_nht,
 			if ((zvrf = vrf->info) != NULL) {
 				vty_out(vty, "\nVRF %s:\n", zvrf_name(zvrf));
 				zebra_print_rnh_table(zvrf_id(zvrf), afi, vty,
-						      RNH_NEXTHOP_TYPE);
+						      rtype, NULL);
 			}
 		return CMD_SUCCESS;
 	}
 	if (vrf_name)
 		VRF_GET_ID(vrf_id, vrf_name, false);
 
-	zebra_print_rnh_table(vrf_id, afi, vty, RNH_NEXTHOP_TYPE);
-	return CMD_SUCCESS;
-}
-
-DEFPY (show_ip_import_check,
-       show_ip_import_check_cmd,
-       "show <ip$ipv4|ipv6$ipv6> import-check [vrf NAME$vrf_name|vrf all$vrf_all]",
-       SHOW_STR
-       IP_STR
-       IP6_STR
-       "IP import check tracking table\n"
-       VRF_CMD_HELP_STR
-       VRF_ALL_CMD_HELP_STR)
-{
-	afi_t afi = ipv4 ? AFI_IP : AFI_IP6;
-	vrf_id_t vrf_id = VRF_DEFAULT;
-
-	if (vrf_all) {
-		struct vrf *vrf;
-		struct zebra_vrf *zvrf;
-
-		RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name)
-			if ((zvrf = vrf->info) != NULL) {
-				vty_out(vty, "\nVRF %s:\n",
-					zvrf_name(zvrf));
-				zebra_print_rnh_table(zvrf_id(zvrf), afi, vty,
-						      RNH_IMPORT_CHECK_TYPE);
-			}
-		return CMD_SUCCESS;
-	}
-	if (vrf_name)
-		VRF_GET_ID(vrf_id, vrf_name, false);
+	memset(&prefix, 0, sizeof(prefix));
+	if (addr) {
+		if (afi == AFI_IP) {
+			prefix.family = AF_INET;
+			prefix.prefixlen = 32;
+			prefix.u.prefix4 = addr->sin.sin_addr;
+		} else if (afi == AFI_IP6) {
+			prefix.family = AF_INET6;
+			prefix.prefixlen = 128;
+			prefix.u.prefix6 = addr->sin6.sin6_addr;
+		}
```

Comment:
nit: I believe we can simplify this code by using `sockunion2hostprefix()`.

Reply:
I'm not sure I follow.  sockunion2hostprefix takes a struct sockunion * and converts it to a prefix.  but I have a const char * that needs to be translated

Reply:
Isn't `addr` a sockunion here?

clippy creates sockunions for `<A.B.C.D|X:X::X:X>` parameters: http://docs.frrouting.org/projects/dev-guide/en/latest/cli.html?highlight=DEFPY#type-rules

Reply:
yeah I'm being silly. I thought I had a char *.  fixed